common/types/mapper/proto/api.go

```diff
@@ -5566,22 +5566,25 @@ func FromActiveClusters(t *types.ActiveClusters) *apiv1.ActiveClusters {
 		return nil
 	}
 
+	// Convert AttributeScopes to both the new format and backward-compatible regionToCluster
 	var regionToCluster map[string]*apiv1.ActiveClusterInfo
-	if len(t.ActiveClustersByRegion) > 0 {
-		regionToCluster = make(map[string]*apiv1.ActiveClusterInfo)
-		for region, cluster := range t.ActiveClustersByRegion {
-			regionToCluster[region] = &apiv1.ActiveClusterInfo{
-				ActiveClusterName: cluster.ActiveClusterName,
-				FailoverVersion:   cluster.FailoverVersion,
-			}
-		}
-	}
-
 	var activeClustersByClusterAttribute map[string]*apiv1.ClusterAttributeScope
+
 	if t.AttributeScopes != nil {
 		activeClustersByClusterAttribute = make(map[string]*apiv1.ClusterAttributeScope)
 		for scopeType, scope := range t.AttributeScopes {
 			activeClustersByClusterAttribute[scopeType] = FromClusterAttributeScope(&scope)
+
+			// For backward compatibility, populate regionToCluster from the "region" scope
+			if scopeType == types.DefaultAttributeScopeType {
```

Comment:
I think we don't need this backward compatibility.

Reply:
yeah fair, I can remove